registry/storage/blobcachemetrics.go

```diff
@@ -0,0 +1,60 @@
+package storage
+
+import (
+	"expvar"
+	"sync/atomic"
+
+	"github.com/docker/distribution/registry/storage/cache"
+)
+
+type blobStatCollector struct {
+	metrics cache.Metrics
+}
+
+func (bsc *blobStatCollector) Hit() {
```

Comment:
We are going down a slippery slope here. Let's make a task to get the metrics situation under control.

I do like this interface. We should have a common metrics productions, for things like caches, in addition to standard guages and timers. @discordianfish